### PR TITLE
feat(command/analyzer): add diagnostics to analyzer command

### PR DESCRIPTION
**Pull request by burntcarrot**
Adds diagnostics support for the `deepsource analyzer verify` command.

Pretty prints diagnostics to the console.

Screencast (updated): https://asciinema.org/a/XETxp6XTNdN3cjqJMfzR7em3i

### CODE REVIEW

**Review by siddhant-deepsource — CHANGES_REQUESTED** [review]
This looks good. Will be testing it once locally on my end as well. Have made some comments. Thanks!

File: analyzers/diagnostics/diagnostics.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	return string(content), nil
+}
+
+// getDiagnosticsFromFile uses the file content to return diagnostics with metadata like line number, content, etc.
+func getDiagnosticsFromFile(fileContent string, errors []validator.ErrorMeta) []Diagnostic {
+	diagnostics := []Diagnostic{}
+
+	lines := strings.Split(string(fileContent), "\n")
+	if len(lines) < 3 {
```
[the remark this answers is not in the document]
Comment (burntcarrot) [reply analyzers/diagnostics/diagnostics.go]: It should work normally. I've updated the tests to test for this behavior.

File: analyzers/diagnostics/diagnostics.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	}
+	match = strings.TrimSpace(match)
+
+	if strings.HasPrefix(match, "#") && strings.Contains(match, field) {
+		// Return true if the current line is a comment, and contains the field.
+		// Useful for scenarios where the user tends to comment out fields.
+		return true
+	} else if strings.Contains(match, field) {
+		// If the current is not a comment, but if it contains the field, return true.
+		return true
```
Comment (siddhant-deepsource) [inline analyzers/diagnostics/diagnostics.go:144]: These conditions can be clubbed using `||` operator.

Comment (siddhant-deepsource) [inline analyzers/diagnostics/diagnostics.go]:
```suggestion
  // Return true if the current line is a comment, and contains the field.
	// Useful for scenarios where the user tends to comment out fields.
	// If the current is not a comment, but if it contains the field, return true.
	if strings.HasPrefix(match, "#") && strings.Contains(match, field) || strings.Contains(match, field) {
		return true
	}
```

Comment (siddhant-deepsource) [reply analyzers/diagnostics/diagnostics.go]: Added as suggestion.

File: analyzers/diagnostics/diagnostics.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	// Get index for the "field" named group.
+	matches := exp.FindStringSubmatch(line)
+	idx := exp.SubexpIndex("field")
+
+	// Capture match.
+	match := ""
+	if len(matches) != 0 && idx != -1 {
+		match = matches[idx]
+	}
+	match = strings.TrimSpace(match)
```
Comment (siddhant-deepsource) [inline analyzers/diagnostics/diagnostics.go:136]: What is `idx`? Can you please add a helpful comment for what is being done here?

**Review by siddhant-deepsource — CHANGES_REQUESTED** [review]

File: analyzers/diagnostics/diagnostics.go (hunk 2 above)
Comment (siddhant-deepsource) [reply analyzers/diagnostics/diagnostics.go]: Fixed in [1fb50b8](https://github.com/deepsourcelabs/cli/pull/169/commits/1fb50b8cbe89537548b46b996d7b69fd84f41c95). Resolving.

Comment (burntcarrot) [reply analyzers/diagnostics/diagnostics.go]: I tried it out, but it doesn't work in some cases. The tests fail in https://github.com/deepsourcelabs/cli/pull/169/commits/1fb50b8cbe89537548b46b996d7b69fd84f41c95.

File: analyzers/diagnostics/diagnostics.go (hunk 3 above)
Comment (burntcarrot) [reply analyzers/diagnostics/diagnostics.go]: Resolved in https://github.com/deepsourcelabs/cli/pull/169/commits/61385c434b0e5fb9df909d816ff797e8101b16b8.

Comment (burntcarrot) [reply analyzers/diagnostics/diagnostics.go]: Resolved in https://github.com/deepsourcelabs/cli/pull/169/commits/e9709d9e23dafe50fe102011140d1037f6435a6e.

**Review by siddhant-deepsource — APPROVED** [review]
Looks good. Lets get it integrated with issue TOMLs as well. Thanks! 🚀

**Pull request merged**